Failed to build with Linux v6.11

Reported by CI, on a Linux v6.11 host:
```
make -C /lib/modules/6.11.0-1012-azure/build M=/home/runner/work/vwifi/vwifi modules
  cc -std=gnu99 -Wno-declaration-after-statement -o vwifi-tool vwifi-tool.c
  make[1]: Entering directory '/usr/src/linux-headers-6.11.0-1012-azure'
  warning: the compiler differs from the one used to build the kernel
    The kernel was built by: x86_64-linux-gnu-gcc-13 (Ubuntu 13.3.0-6ubuntu2~24.04) 13.3.0
    You are using:           gcc-13 (Ubuntu 13.3.0-6ubuntu2~24.04) 13.3.0
    CC [M]  /home/runner/work/vwifi/vwifi/vwifi.o
  /home/runner/work/vwifi/vwifi/vwifi.c: In function ‘vwifi_virtio_init_vqs’:
  /home/runner/work/vwifi/vwifi/vwifi.c:3170:60: error: passing argument 4 of ‘virtio_find_vqs’ from incompatible pointer type [-Werror=incompatible-pointer-types]
   3170 |     return virtio_find_vqs(vdev, VWIFI_NUM_VQS, vwifi_vqs, callbacks, names,
        |                                                            ^~~~~~~~~
        |                                                            |
        |                                                            void (**)(struct virtqueue *)
  In file included from /home/runner/work/vwifi/vwifi/vwifi.c:11:
  ./include/linux/virtio_config.h:223:43: note: expected ‘struct virtqueue_info *’ but argument is of type ‘void (**)(struct virtqueue *)’
    223 |                     struct virtqueue_info vqs_info[],
        |                     ~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~
  /home/runner/work/vwifi/vwifi/vwifi.c:3170:71: error: passing argument 5 of ‘virtio_find_vqs’ from incompatible pointer type [-Werror=incompatible-pointer-types]
   3170 |     return virtio_find_vqs(vdev, VWIFI_NUM_VQS, vwifi_vqs, callbacks, names,
        |                                                                       ^~~~~
        |                                                                       |
        |                                                                       const char **
  ./include/linux/virtio_config.h:224:42: note: expected ‘struct irq_affinity *’ but argument is of type ‘const char **’
    224 |                     struct irq_affinity *desc)
        |                     ~~~~~~~~~~~~~~~~~~~~~^~~~
  /home/runner/work/vwifi/vwifi/vwifi.c:3170:12: error: too many arguments to function ‘virtio_find_vqs’
   3170 |     return virtio_find_vqs(vdev, VWIFI_NUM_VQS, vwifi_vqs, callbacks, names,
        |            ^~~~~~~~~~~~~~~
  ./include/linux/virtio_config.h:221:5: note: declared here
    221 | int virtio_find_vqs(struct virtio_device *vdev, unsigned int nvqs,
        |     ^~~~~~~~~~~~~~~
  cc1: some warnings being treated as errors
  make[3]: *** [scripts/Makefile.build:244: /home/runner/work/vwifi/vwifi/vwifi.o] Error 1
  make[2]: *** [/usr/src/linux-headers-6.11.0-[101](https://github.com/sysprog21/vwifi/actions/runs/14741735868/job/41392812159#step:4:102)2-azure/Makefile:1932: /home/runner/work/vwifi/vwifi] Error 2
  make[1]: *** [Makefile:224: __sub-make] Error 2
  make[1]: Leaving directory '/usr/src/linux-headers-6.11.0-1012-azure'
  make: *** [Makefile:11: kmod] Error 2
  cpp/autobuilder: trying to run make [current dir: /home/runner/work/vwifi/vwifi]
  make -C /lib/modules/6.11.0-1012-azure/build M=/home/runner/work/vwifi/vwifi modules
  make[1]: Entering directory '/usr/src/linux-headers-6.11.0-1012-azure'
  warning: the compiler differs from the one used to build the kernel
    The kernel was built by: x86_64-linux-gnu-gcc-13 (Ubuntu 13.3.0-6ubuntu2~24.04) 13.3.0
    You are using:           gcc-13 (Ubuntu 13.3.0-6ubuntu2~24.04) 13.3.0
    CC [M]  /home/runner/work/vwifi/vwifi/vwifi.o
  /home/runner/work/vwifi/vwifi/vwifi.c: In function ‘vwifi_virtio_init_vqs’:
  /home/runner/work/vwifi/vwifi/vwifi.c:3170:60: error: passing argument 4 of ‘virtio_find_vqs’ from incompatible pointer type [-Werror=incompatible-pointer-types]
   3170 |     return virtio_find_vqs(vdev, VWIFI_NUM_VQS, vwifi_vqs, callbacks, names,
        |                                                            ^~~~~~~~~
        |                                                            |
        |                                                            void (**)(struct virtqueue *)
  In file included from /home/runner/work/vwifi/vwifi/vwifi.c:11:
  ./include/linux/virtio_config.h:223:43: note: expected ‘struct virtqueue_info *’ but argument is of type ‘void (**)(struct virtqueue *)’
    223 |                     struct virtqueue_info vqs_info[],
        |                     ~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~
  /home/runner/work/vwifi/vwifi/vwifi.c:3170:71: error: passing argument 5 of ‘virtio_find_vqs’ from incompatible pointer type [-Werror=incompatible-pointer-types]
   3170 |     return virtio_find_vqs(vdev, VWIFI_NUM_VQS, vwifi_vqs, callbacks, names,
        |                                                                       ^~~~~
        |                                                                       |
        |                                                                       const char **
  ./include/linux/virtio_config.h:224:42: note: expected ‘struct irq_affinity *’ but argument is of type ‘const char **’
    224 |                     struct irq_affinity *desc)
        |                     ~~~~~~~~~~~~~~~~~~~~~^~~~
  /home/runner/work/vwifi/vwifi/vwifi.c:3170:12: error: too many arguments to function ‘virtio_find_vqs’
   3170 |     return virtio_find_vqs(vdev, VWIFI_NUM_VQS, vwifi_vqs, callbacks, names,
        |            ^~~~~~~~~~~~~~~
  ./include/linux/virtio_config.h:221:5: note: declared here
    221 | int virtio_find_vqs(struct virtio_device *vdev, unsigned int nvqs,
        |     ^~~~~~~~~~~~~~~
  cc1: some warnings being treated as errors
  make[3]: *** [scripts/Makefile.build:244: /home/runner/work/vwifi/vwifi/vwifi.o] Error 1
```

--- a/vwifi.c
+++ b/vwifi.c
@@ -3167,8 +3167,23 @@ static int vwifi_virtio_init_vqs(struct virtio_device *vdev)
         [VWIFI_VQ_TX] = "tx",
     };
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 11, 0)
+    struct virtqueue_info vqs_info[VWIFI_NUM_VQS];
+    /* Set ctx to false to maintain compatibility with legacy
+     * virtio_find_vqs() behavior, which implicitly passed NULL for ctx.
+     */
+    for (int i = 0; i < VWIFI_NUM_VQS; i++) {
+        vqs_info[i].callback = callbacks[i];
+        vqs_info[i].name = names[i];
+        vqs_info[i].ctx = false;
+    }
+
+    return virtio_find_vqs(vdev, VWIFI_NUM_VQS, vwifi_vqs, vqs_info, NULL);
+
+#else
     return virtio_find_vqs(vdev, VWIFI_NUM_VQS, vwifi_vqs, callbacks, names,
                            NULL);
+#endif
 }
 
 static void vwifi_virtio_fill_vq(struct virtqueue *vq, u8 vnet_hdr_len)

